modules/calcite/src/main/java/org/apache/ignite/internal/processors/query/calcite/hint/HintsConfig.java

```diff
@@ -77,9 +81,38 @@ private HintsConfig() {
     public static HintStrategyTable buildHintTable() {
         HintStrategyTable.Builder b = HintStrategyTable.builder().errorHandler(Litmus.IGNORE);
 
+        RelOptRule[] disabledRulesTpl = new RelOptRule[0];
+
         Arrays.stream(HintDefinition.values()).forEach(hintDef ->
-            b.hintStrategy(hintDef.name(), HintStrategy.builder(hintDef.predicate()).build()));
+            b.hintStrategy(hintDef.name(), HintStrategy.builder(hintPredicate(hintDef))
+                .excludedRules(hintDef.disabledRules().toArray(disabledRulesTpl)).build()));
 
         return b.build();
     }
+
+    /**
+     * Adds hint options checker to {@link HintPredicate} if {@code hintDef} has rules to exclude.
+     *
+     * @return Hint predicate.
+     */
+    private static HintPredicate hintPredicate(HintDefinition hintDef) {
+        if (F.isEmpty(hintDef.disabledRules()))
```

Comment:
Looks like just masking of ORDERED_JOINS (perhaps it's not correct condition for all hints with disabled rules in future). And why do we need it only for ORDERED_JOINS?

Reply:
Because hint might not be used in any rule. Some hints may only bring dynamically new rules to the planner or mark some rules as skippable. Or use both approaches. Sice we do not use Calcite's `Litmus` and `HintErrorLogger` (they has own logger or requre dynamic logger initialization), we do not do general validation of hint params in this case. Usually, we do that in `HintUtils#filterHints()` . While the hint predicate is mandatory to call by Calcite to assing a hint to a rel node. Without wrapping of the hint predicate, `JoinOrderHintsPlannerTest#testWrongParams()` won't work.